.github/workflows/publish.yml

```diff
@@ -0,0 +1,190 @@
+    - name: Determine whether releasing stable branch for pushed tag
+      id: set-is-stable-branch
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |
+          var resp
+          try {
+            resp = await github.rest.git.getRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: "heads/${{ steps.set-stable-branch.outputs.result }}",
+            })
+          }
+          catch(err) {
+            console.log("false (stable branch does not exist: "+err+")")
+            return false
+          }
+          const result = (resp.data.object.sha == "${{ github.sha }}")
+          console.log(result)
+          return result
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Check released commit to be master branch or stable branch for pushed tag
+      run: |
+        if [[ ${{ steps.set-is-master-branch.outputs.result }} == 'false' && ${{ steps.set-is-stable-branch.outputs.result }} == 'false' ]]; then
+          echo "Released commit is not 'master' or '${{ steps.set-stable-branch.outputs.result }}' branch";
+          false;
+        fi
+    - name: Set update version
+      id: set-update-version
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |
+          const result = "${{ steps.set-tag.outputs.result }}".match("[0-9]+\.[0-9]+\.([0-9]+)")[1]
+          console.log(result)
+          return result
+    - name: Check update version to be 0 when releasing master branch
+      if: ${{ steps.set-is-master-branch.outputs.result == 'true' }}
+      run: |
+        if [[ ${{ steps.set-update-version.outputs.result }} != '0' ]]; then
+          echo "Update version '${{ steps.set-update-version.outputs.result }}' in tag '${{ steps.set-tag.outputs.result }}' is invalid (must be 0 when releasing master branch)";
+          false;
+        fi
+    - name: Check update version to be non-0 when releasing stable branch for pushed tag
+      if: ${{ steps.set-is-stable-branch.outputs.result == 'true' }}
+      run: |
+        if [[ ${{ steps.set-update-version.outputs.result }} == '0' ]]; then
+          echo "Update version '${{ steps.set-update-version.outputs.result }}' in tag '${{ steps.set-tag.outputs.result }}' is invalid (must be non-0 when releasing stable branch for pushed tag)";
+          false;
+        fi
+
+    #-------- Setup of work environment
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.12"
+    - name: Development setup
+      run: |
+        make develop
+
+    #-------- Publishing of package
+    - name: Build the distribution
+      run: |
+        make build
+    - name: Display the distribution directory
+      run: |
+        ls -l dist
+    # - name: Publish distribution to TestPyPI
```

Comment:
This group of comments referencing TestPyPi confuses me.

Reply:
It is there so it can be enabled for testing or debugging the release process.